homeassistant/components/habitica/sensor.py

```diff
@@ -77,3 +129,55 @@ def state(self):
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
         return self._sensor_type.unit
+
+
+class HabitipyTaskSensor(Entity):
+    """A Habitica task sensor."""
+
+    def __init__(self, name, task_name, updater):
+        """Initialize a generic Habitica task."""
+        self._name = name
+        self._task_name = task_name
+        self._task_type = TASKS_TYPES[task_name]
+        self._state = None
+        self._updater = updater
+
+    async def async_update(self):
+        """Update Condition and Forecast."""
+        await self._updater.update()
+        all_tasks = self._updater.tasks
+        for element in self._task_type.path:
+            tasks_length = len(all_tasks[element])
+        self._state = tasks_length
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return self._task_type.icon
+
+    @property
+    def name(self):
+        """Return the name of the task."""
+        return f"{habitica.DOMAIN}_{self._name}_{self._task_name}"
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of all user tasks."""
+        if self._updater.tasks is not None:
+            all_tasks = self._updater.tasks
+            for element in self._task_type.path:
+                tasks = all_tasks[element]
+            attrs = {}
+            for task in tasks:
+                attrs[task["id"]] = task
```

Comment:
What kind of values are the tasks?

Reply:
It's a list of dictionaries. See a sample [here](https://pastebin.com/cZWtwsNU)

Reply:
Ok, so one task is a dictionary with `challenge`, `group`, `up`, `down` items etc.?

Reply:
Yes, right, @MartinHjelmare